documentation/CAMARA-OIDC-profile.md

```diff
@@ -0,0 +1,1022 @@
+
+Communication with the Backchannel Authentication Endpoint MUST utilize TLS.
+
+CIBA allows the Client to get the authentication result in three ways: poll, ping, or push. This profile only allows poll mode. In the Poll mode, the authentication result is retrieved by the Client by polling the OP's token endpoint using the new grant type.
+
+### Authentication Request
+
+An authentication request is composed of the following parameters and MAY contain additional parameters defined by extension or profile:
+
+* **scope**
+
+  REQUIRED. A space delimited and a case-sensitive list of ASCII strings for OAuth 2.0 scope values. Consistent with that, CIBA authentication requests MUST therefore contain the `openid` scope value. More information about scope values as described in Scope Parameter Section.
+
+
+* **login_hint**
+
+  REQUIRED. A hint to the OpenID Provider regarding the end-user for whom authentication is being requested. The value may contain a phone number, ip, etc., which identifies the end-user to the Authorization Server. The value may be directly collected from the user by the Client before requesting authentication at the Authorization Server, for example, but may also be obtained by other means.
+
+  The `login_hint` value will follow the format: `identifier_type:identifier_value`.
+
+  The defined values are for each identifier type:
+
+    * **_tel_**
+
+      For phone numbers. The `login_hint` must be a tel URI as defined in [RFC 3966](https://www.rfc-editor.org/info/rfc3966) for global phone numbers without visual separators in [E.164](https://www.itu.int/rec/T-REC-E.164-201011-I/en) format. For example, `tel:+34666666666`.
+
+    * **_ipport_**
+
+      For IPv4 and IPv6 addresses, that can optionally include a port. For example, `ipport:80.90.34.2:16790`, `ipport:80.90.34.2`, `ipport:[2001:db8::1]:8080` or `ipport:[2001:db8::1]`.
+
+
+Because in the CIBA flow, the Authorization Server does not have an interaction with the end-user through the consumption device, it is REQUIRED that the Client provides a `login_hint`.
+
+An authentication request is made using the HTTP POST method with the aforementioned parameters in the `application/x-www-form-urlencoded` format and a character encoding of UTF-8 in the HTTP request entity-body. When applicable, additional parameters required by the given client authentication method are also included (e.g. JWT assertion-based client authentication uses `client_assertion` and `client_assertion_type`).
+
+The client MUST authenticate with the authorization server as described in [Client Authentication Section](#client-authentication).
```

Comment:
If the client authentication method is `private_key_jwt`, then it should be mandated that the authorisation request be signed (as defined [here](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#rfc.section.7.1.1)). An example of a signed request should be provided.

Reply:
As far as I know, the use of the `request` parameter is not approved by all of us. This profile only includes what is already agreed, but we could add it in the future if there is consensus. Anyway, the standard does not say that it is mandatory to pass the `request` parameter if the authentication method is `private_key_jwt`. In fact a few lines above the link you sent there is an example of `private_key_jwt` authentication without request object parameter.

Reply:
Sure, but if we were only following what the standard says, then we would not need this profile.

The purpose of the profile is to **reduce** the number of options available. But here we still have two - signed or unsigned requests, applicable to all allowed client authorisation schemes. But if the API provider already knows the client public key for `private_key_jwt`, then verifying a signed request is trivial.

Reply:
> the `request` object parameter will be redundant for security purposes

No, not redundant. Signed authorisation requests ensure the integrity of the request, and prevent it being modified by a MITM (or, more accurately, allow such modifications to be detected).

Now it could be argued that TLS will prevent such attacks and thus render them purely theoretical. But nonetheless signed requests are defined as an option by CIBA and, given that there is no good reason not to support signed requests if the client public key is anyway known, will likely be mandated by at least some API providers.

So I think signed authorisation requests should be mandatory (given that `client_secret_basic` is no longer being proposed as a valid client authentication scheme). That would remove another CIBA option, thus increasing the likelihood of the API provider and consumer supporting a compatible authorisation flow.

Reply:
Sure, but signed or unsigned authorisation requests both remain as options. One of those options should be removed. I propose to mandate signed requests.

Reply:
@eric-murray , you are using the MITM argument to refute my claim that it could be used in the same way to not trust the public keys exposed in the jwks_uri.

Reply:
Well, if a MITM is intercepting **all** traffic between the API consumer and provider at HTTP level, then indeed that client is hopelessly compromised. But that is an even more theoretical attack. I think we have to make the assumption that the API provider knows the correct client public key.

But the **key point** remains that this profile allows both signed and unsigned authorisation requests. Only one of these options should be allowed, and my preference is signed authorisation requests. As long as signed authorisation requests remain an option, some API providers will mandate it because their own internal security rules will not allow accepting unsigned requests when they could be signed.

Reply:
I am not particularly keen on rejecting the inclusion of the `request` parameter in the CIBA flow. We can include this requirement in future versions of the profile if there is consensus. 

Nice theoretical discussion. :smiley: